### PR TITLE
Implement card cycling in previewer

### PR DESCRIPTION
**Pull request by hssm**
Implements https://github.com/ankidroid/Anki-Android/issues/1381. Works the same way as the desktop version.

The button colors could still use some work.

### CODE REVIEW

**Comment by timrae** [comment]
@hssm 
Is there a reason why you merged your own PR rather than waiting for code review? I'm highly guilty of this as well, especially around release time, but we probably shouldn't really be doing it.

I wonder if it would be productive to introduce some more clear and publicly visible / transparent guidelines around the development process which we actually follow. For example, recently I'm learning ZeroMQ for work and was wondering if their [C4 process](http://rfc.zeromq.org/spec:22/C4/) would work for us.

What do others think about C4 in specific, and/or the idea in general?

File: AnkiDroid/src/main/java/com/ichi2/anki/Previewer.java [hunk 1]
```diff
@@ -45,7 +53,7 @@ protected void onCreate(Bundle savedInstanceState) {
     @Override
     protected void onCollectionLoaded(Collection col) {
         super.onCollectionLoaded(col);
-        mCurrentCard = col.getCard(mCurrentCardId);
+        mCurrentCard = col.getCard(mCardList[mIndex]);
```
Comment (timrae) [inline AnkiDroid/src/main/java/com/ichi2/anki/Previewer.java]: There should probably be an out of bounds check here

File: AnkiDroid/src/main/java/com/ichi2/anki/Previewer.java [hunk 2]
```diff
@@ -25,15 +25,23 @@
...
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         Timber.d("onCreate()");
-        mCurrentCardId=getIntent().getLongExtra("currentCardId", -1);
-        if (mCurrentCardId == -1) {
-            Timber.e("Previewer started without a valid card ID");
+        mCardList = getIntent().getLongArrayExtra("cardList");
+        mIndex = getIntent().getIntExtra("index", -1);
+        if (mCardList.length == 0 || mIndex == -1) {
+            Timber.e("Previewer started without empty card list or invalid index");
```
Comment (timrae) [inline AnkiDroid/src/main/java/com/ichi2/anki/Previewer.java]: I guess it should be

Previewer started *with* empty card list or invalid index

:)

**Comment by hssm** [comment]
Sorry I got the impression you didn't have time to look over all the little things (although I guess this isn't really that little). I was also scrambling to get this completed and merged so I can make the new hotfix branch since this was all I intended to put into it.

I think a process will just get in the way. We don't really gain much and I honestly can't be bothered.

**Pull request merged**